src/shared_modules/utils/rocksDBWrapper.hpp

```diff
@@ -0,0 +1,190 @@
+         * @return bool False if the key was not found.
+         *
+         */
+        bool get(const std::string& key, std::string& value)
+        {
+            if (key.empty())
+            {
+                throw std::invalid_argument("Key is empty");
+            }
+
+            const auto status {m_db->Get(rocksdb::ReadOptions(), key, &value)};
+            if (status.IsNotFound())
+            {
+                std::cerr << "Key not found: " << key << '\n';
+                return false;
+            }
+            else if (!status.ok())
+            {
+                throw std::runtime_error("Error getting data: " + status.ToString());
+            }
+            return true;
+        }
+
+        /**
+         * @brief Delete a key-value pair from the database.
+         *
+         * @param key Key to delete.
+         */
+        void delete_(const std::string& key) // NOLINT
+        {
+            if (key.empty())
+            {
+                throw std::invalid_argument("Key is empty");
+            }
+
+            const auto status {m_db->Delete(rocksdb::WriteOptions(), key)};
+            if (!status.ok())
+            {
+                throw std::runtime_error("Error deleting data: " + status.ToString());
+            }
+        }
+
+        /**
+         * @brief Get the last key-value pair from the database.
+         *
+         * @return std::pair<std::string, std::string> Last key-value pair.
+         *
+         * @note The first element of the pair is the key, the second element is the value.
+         */
+        std::pair<std::string, std::string> getLastKeyValue()
+        {
+            std::unique_ptr<rocksdb::Iterator> it(m_db->NewIterator(rocksdb::ReadOptions()));
+
+            it->SeekToLast();
+            if (it->Valid())
+            {
+                return {it->key().ToString(), it->value().ToString()};
+            }
+
+            throw std::runtime_error {"Error getting last key-value pair"};
+        }
+
+        /**
+         * @brief Delete all key-value pairs from the database.
+         */
+        void deleteAll()
+        {
+            rocksdb::WriteBatch batch;
+            std::unique_ptr<rocksdb::Iterator> it(m_db->NewIterator(rocksdb::ReadOptions()));
+            for (it->SeekToFirst(); it->Valid(); it->Next())
+            {
+                batch.Delete(it->key());
+            }
+
+            const auto status {m_db->Write(rocksdb::WriteOptions(), &batch)};
+            if (!status.ok())
+            {
+                throw std::runtime_error("Error deleting data: " + status.ToString());
+            }
+        }
+
+        /**
+         * @brief Get an iterator to the database.
+         *
+         * @return std::unique_ptr<rocksdb::Iterator> Iterator to the database.
+         */
+        std::shared_ptr<rocksdb::Iterator> getIterator()
+        {
+            return std::shared_ptr<rocksdb::Iterator>(m_db->NewIterator(rocksdb::ReadOptions()));
```

Comment:
This function [returns a new shared pointer every time is called](https://godbolt.org/z/s71cae8eK), we can use one shared pointer and increase its internal counter instead of creating a new pointer when we need the iterator.

Reply:
We can probably take this out, that function or it is encapsulating the iteration part of rocksdb.